### PR TITLE
docs - new pxf IGNORE_MISSING_PATH option

### PR DESCRIPTION
**Pull request by lisakowen**
pxf introduces the IGNORE_MISSING_PATH option.  in this PR:
- add new section "About PXF External Table Child Partitions" to the troubleshooting page.  i added it here because it seems to me that this isn't that common of a scenario.  i could be convinced to add it elsewhere.
- this new option is available for many, but not all, of the profiles.  i added an entry for the option on the hdfs:text page, in the external table parameters table.  i can also add this option/description to the external table parameters tables of all applicable profiles.   but just wondering if we want to do this?  if the option won't typically be used, the description could introduce confusion.  interested in your thoughts.

doc review site link:  http://docs-lisa-newpxfopt.cfapps.io/7-0/pxf/troubleshooting_pxf.html#pxf-tblpart

### CODE REVIEW

File: gpdb-doc/markdown/pxf/troubleshooting_pxf.html.md.erb [hunk 1]
```diff
@@ -356,3 +356,22 @@ PXF fragment metadata caching is enabled by default. To turn off fragment metada
...
 5. Restart PXF on each Greenplum Database segment host as described in [Restarting PXF](cfginitstart_pxf.html#restart_pxf).
 
+
+## <a id="pxf-tblpart"></a>About PXF External Table Child Partitions
+
+Greenplum Database supports partitioned tables, and permits exchanging a leaf child partition with a PXF external table.
+
+When you read from a partitioned Greenplum table where one or more partitions is a PXF external table and there is no data backing the external table path, PXF returns an error and the query fails. This behavior is not optimal.
+
+The `IGNORE_MISSING_PATH` PXF custom option is a boolean that specifies PXF's action when the external table path is missing or invalid. The default value is `false`, PXF does not ignore a missing path and returns an error. If the external table is a partition of a Greenplum table and you want PXF to ignore a missing path error, set this option to `true`.
```
Comment (oliverralbertini) [inline gpdb-doc/markdown/pxf/troubleshooting_pxf.html.md.erb:366]: minor suggestion:
PXF does not ignore a missing path and returns an error -> if PXF encounters a missing path it returns an error

**Comment by lisakowen** [comment]
@frankgh , @oliverralbertini - thanks for reviewing.  wondering if i should add the option to the external table parameters table for all profiles that support this option - it would look something like the changes to the hdfs_text.html.md.erb file in this PR.  if not, i would remove the option from hdfs_text, and the only reference to it would be on the troubleshooting page.  thoughts?

**Comment by frankgh** [comment]
@lisakowen I thought about that too, and didn't want to mention it. But I think it makes sense for consistency. Can you add it to the other profiles too?

**Comment by lisakowen** [comment]
@frankgh @oliverralbertini - i added info about the new option to relevant profiles.  note that some of the object store topics refer back to the hdfs topic for custom options.  would you mind taking a look at the changes in the newest commit?

i also repushed the review site, may need to reload.

**Comment by oliverralbertini** [comment]
> @lisakowen I thought about that too, and didn't want to mention it. But I think it makes sense for consistency. Can you add it to the other profiles too?

I agree, I think it's better to have this information on all the profiles, not just in the troubleshooting.

File: gpdb-doc/markdown/pxf/troubleshooting_pxf.html.md.erb [hunk 2]
```diff
@@ -356,3 +356,22 @@ PXF fragment metadata caching is enabled by default. To turn off fragment metada
 
 5. Restart PXF on each Greenplum Database segment host as described in [Restarting PXF](cfginitstart_pxf.html#restart_pxf).
 
+
+## <a id="pxf-tblpart"></a>About PXF External Table Child Partitions
+
+Greenplum Database supports partitioned tables, and permits exchanging a leaf child partition with a PXF external table.
+
+When you read from a partitioned Greenplum table where one or more partitions is a PXF external table and there is no data backing the external table path, PXF returns an error and the query fails. This behavior is not optimal.
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply gpdb-doc/markdown/pxf/troubleshooting_pxf.html.md.erb]: thanks for reviewing, david.  the default behaviour (error on missing path) is optimal for the main use case - reading data from an external data store and the path doesn't exist.  if i understand it correctly, the new option is really for the special case where the pxf external table is a child partition.  wouldn't want an empty partition to cause a query of the parent table to fail, so the user would want pxf to ignore the missing path error in this case.  i updated the text to try to make this (special case) more obvious.

[the remark this answers is not in the document]
Comment (dyozie) [reply gpdb-doc/markdown/pxf/troubleshooting_pxf.html.md.erb]: Thanks - that rewrite helps a lot!

**Pull request merged**